test/middleware/express.test.ts

```diff
@@ -0,0 +1,341 @@
+                    // successful request calls next without any arguments.
+                    middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+                    expect(nextFunction).toBeCalledTimes(1);
+                    expect(nextFunction).toBeCalledWith();
+                });
+
+                test('Multiple valid requests at > 1 second intervals', () => {
+                    for (let i = 0; i < 3; i++) {
+                        const next: NextFunction = jest.fn();
+                        middleware(complexRequest as Request, mockResponse as Response, next);
+                        expect(next).toBeCalledTimes(1);
+                        expect(next).toBeCalledWith();
+
+                        // advance the timers by 1 second for the next request
+                        jest.advanceTimersByTime(1000);
+                    }
+                });
+
+                test('Multiple valid requests at within one second', () => {
+                    for (let i = 0; i < 3; i++) {
+                        const next: NextFunction = jest.fn();
+                        middleware(complexRequest as Request, mockResponse as Response, next);
+                        expect(next).toBeCalledTimes(1);
+                        expect(next).toBeCalledWith();
+
+                        // advance the timers by 1 second for the next request
+                        jest.advanceTimersByTime(20);
+                    }
+                });
+            });
+
+            describe('BLOCKS requests', () => {
+                test('A single request that exceeds capacity', () => {
+                    complexRequest = {
+                        // complexity should be 12 if 'first' is accounted for.
+                        // scalars: 1, droid: 1, reviews (5 * (1 Review, 1 episode))
+                        query: {
+                            query: `Query {
+                            scalars: {
+                                num
+                            }
+                            droid(id: 1) {
+                                name
+                            }
+                            reviews(episode: 'NEWHOPE', first: 5) {
+                                episode 
+                                stars
+                                commentary
+                            }
+                        `,
+                        },
+                    };
+                    const expectedResponse = {
+                        status: 429,
+                    };
+
+                    middleware(mockRequest as Request, mockResponse as Response, nextFunction);
+                    expect(mockResponse.status).toBe(429);
+                    expect(nextFunction).not.toBeCalled();
+
+                    // FIXME: There are multiple functions to send a response
+                    // json, send html, sendStatus etc. How do we check at least one was called
```

Comment:
I think our middleware is only invoking the `send` method with the default status of `429`. What json data or html files would we be responsible for sending in this middelware?

Reply:
Can expand on these tests during implementation once more details are finalized